src/Compilers/Core/MSBuildTask/Vbc.cs

```diff
@@ -651,12 +651,9 @@ protected override void LogEventsFromTextOutput(string singleLine, MessageImport
         /// Given a string, parses it to find out whether it's an error or warning and, if so,
         /// make sure it's validated properly.  
         /// </summary>
-        /// <comments>
-        /// INTERNAL FOR UNITTESTING ONLY
-        /// </comments>
         /// <param name="singleLine">The line to parse</param>
         /// <param name="messageImportance">The MessageImportance to use when reporting the error.</param>
-        internal void ParseVBErrorOrWarning(string singleLine, MessageImportance messageImportance)
+        private void ParseVBErrorOrWarning(string singleLine, MessageImportance messageImportance)
```

Comment:
Agree in general that it's hard to test the various `Tasks` because it becomes more of an integration test with MSBuild. In this particular case though we're effectively altering how VB error messages are parsed in the task. I think we could factor this method a bit, by either adding a `static` method or by just newing up the `Task` in test and setting the properies, and use that to validate this change works (and stays correct going forward).